src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -294,7 +298,8 @@ impl<T: Timestamp + Lattice + Codec64 + Display> AsyncStorageWorker<T> {
                                 .await
                                 .unwrap();
                             let upper = write_handle.fetch_recent_upper().await;
-                            resume_uppers.insert(*id, upper.clone());
+                            resume_uppers
+                                .insert(*id, (upper.clone(), export.data_config.envelope.clone()));
```

Comment:
Let's move the special casing of the envelopes here and also invert the logic to talk about CDCv2 which is the only problematic envelope. The way it's currently written (totally my fault) is very confusing and I had to look at the definition of the envelope enum to understand what this is trying to do.

```suggestion
                            let upper = match export.data_config.envelope {
                                // The CdcV2 envelope must re-ingest everything since the Mz frontier does not have a relation to upstream timestamps.
                                // TODO(petrosagg): move this reasoning to the controller
                                SourceEnvelope::CdcV2 if upper.is_empty() => Antichain::new(),
                                SourceEnvelope::CdcV2 => Antichain::from_elem(Timestamp::minimum()),
                                _ => upper,
                            };
                            resume_uppers.insert(*id, upper.clone());
```

Reply:
Sounds good - that is much cleaner!